synapse/tests/test_datamodel.py

```diff
@@ -124,6 +124,13 @@ async def test_datamodel_dynamics(self):
         modl.delFormProp('foo:foo', 'bar')
         modl.delForm('foo:foo')
 
+        with self.getAsyncLoggerStream('synapse.datamodel') as dstream:
```

Comment:
There is set of taxonomy tests here. They could probably be moved to `test_model_base.py` since the Taxonomy will no longer be defined here.